### ISSUE TITLE
[BUG] (Charting State) Loading a song without difficulty/variation behaves unexpecfully or crashes

### ISSUE THREAD

**Issue by gamerbross** [FunkinCrew/Funkin#2468]
If you had loaded previously a song with another variation (like Erect) or difficulty and then try to load a dfifferent song unexpectedly changes song data or crashes.
(I think only crashes with song templates)

This could be easily fixed by resetting the variation and difficulty when loading a different song.

## Video example: (using song template)

https://github.com/FunkinCrew/Funkin/assets/55158797/af376573-099d-4044-b609-f9dc24ac16d0

## Video example: (loading existing chart)

https://github.com/FunkinCrew/Funkin/assets/55158797/a24ae89c-152f-4f76-bd9b-8a57d6d7d074

**Comment by amyspark-ng** [comment]
nice catch!

**Comment by NotHyper-474** [comment]
0.4.0 and still has not been fixed

**Comment by AbnormalPoof** [comment]
Does this still happen in v0.5.1?

**Comment by NotHyper-474** [comment]
> Does this still happen in v0.5.1?

Yes, and can be replicated if you open a song without a Pico variation as well.

**Comment by NotHyper-474** [comment]
Partial duplicate of #2411

**Comment by Lasercar** [comment]
**_I can't_** reproduce this in 0.5.3/ the development build.

**Comment by VioletSnowLeopard** [comment]
This just in, this bug still exists on the 0.5.3 develop branch:
```
=====================
 Funkin Crash Report
=====================

Generated by: Friday Night Funkin' - v0.5.3 (develop : e83b88d) PROTOTYPE
  Git hash: e83b88d (CLEAN)
System timestamp: 2025-03-26-00-29-35
Driver info: OpenGL Vendor=NVIDIA Corporation Version=4.6.0 NVIDIA 566.07 Renderer=NVIDIA GeForce RTX 4070 Laptop GPU/PCIe/SSE2 GLSL=4.60 NVIDIA
Platform: Windows
Render method: DRAW_TILES

```
I was able to make it occur twice. One for a song without an Erect variation:
```
[lime.utils.Assets] ERROR: There is no BINARY asset with an ID of "songs:assets/songs/2hot/Inst-erect.ogg"
  in lime/utils/Log.hx#37
  in lime/utils/Assets.hx#138
  in lime/utils/Assets.hx#169
  in openfl/utils/Assets.hx#135
  in funkin/Assets.hx#29
  in funkin/ui/debug/charting/handlers/ChartEditorAudioHandler.hx#105
  in funkin/ui/debug/charting/handlers/ChartEditorImportExportHandler.hx#73
  in funkin/ui/debug/charting/dialogs/ChartEditorWelcomeDialog.hx#160
  in haxe/ui/util/EventMap.hx#90
  in haxe/ui/backend/ComponentBase.hx#441
  in haxe/ui/backend/flixel/MouseHelper.hx#201
  in haxe/ui/backend/flixel/MouseHelper.hx#191
  in haxe/ui/backend/flixel/MouseHelper.hx#173
  in haxe/ui/backend/flixel/MouseHelper.hx#103
```

Another for a song without a Pico mix:
```
[lime.utils.Assets] ERROR: There is no BINARY asset with an ID of "songs:assets/songs/thorns/Inst-pico.ogg"
  in lime/utils/Log.hx#37
  in lime/utils/Assets.hx#138
  in lime/utils/Assets.hx#169
  in openfl/utils/Assets.hx#135
  in funkin/Assets.hx#29
  in funkin/ui/debug/charting/handlers/ChartEditorAudioHandler.hx#105
  in funkin/ui/debug/charting/handlers/ChartEditorImportExportHandler.hx#73
  in funkin/ui/debug/charting/dialogs/ChartEditorWelcomeDialog.hx#160
  in haxe/ui/util/EventMap.hx#90
  in haxe/ui/backend/ComponentBase.hx#441
  in haxe/ui/backend/flixel/MouseHelper.hx#201
  in haxe/ui/backend/flixel/MouseHelper.hx#191
  in haxe/ui/backend/flixel/MouseHelper.hx#173
  in haxe/ui/backend/flixel/MouseHelper.hx#103
```

**Comment by VioletSnowLeopard** [comment]
Using debugging with breakpoints, I believe I have found the problematic area in the code. The issue occurs in `ChartEditorImportHandler.hx`. After the `for` loop on line 42, the metadata and chart data for the song you're trying to load are just fine:

![Image](https://github.com/user-attachments/assets/140035f3-63ff-465d-ac7e-196b619de539)

After the call to the `loadSong` function on line 56, however, the variation of the song you had selected previously is erroneously added to the song's metadata. (I had a Pico mix song selected before trying to load Roses):

![Image](https://github.com/user-attachments/assets/0c307c80-0f13-476f-a3e0-a60497c69c78)

This causes the variation to be added to the chart data too after the call to the `sortChartData` function on line 58.

![Image](https://github.com/user-attachments/assets/5e840e01-062b-44b8-8a1d-e93f9a53124f)

Because the issue occurred after the call to the `loadSong` function. I decided to use breakpoints to investigate it. The song metadata and chart data is passed into it correctly:

![Image](https://github.com/user-attachments/assets/c447c75b-4490-48b4-8e93-750e01d42f2c)

These values are fine until this line (line 124) executes:
```haxe
Conductor.instance.instrumentalOffset = state.currentInstrumentalOffset;
```

After that line, the variation for the song you were previously on is added to the metadata of the new song you are trying to load:

![Image](https://github.com/user-attachments/assets/d12eea17-1f03-4827-84bc-1f6718d3f088)

**Comment by NotHyper-474** [comment]
Yep, that happens because when it's retrieving the instrumental offset it will generate a default blank variation if it doesn't exist, I'm not sure if this behavior should be intentional.

![Image](https://github.com/user-attachments/assets/210ef84d-1c23-414d-a6bc-94719f64679e)

**Comment by NotHyper-474** [comment]
Still an issue in 0.6.0
